source/common/http/conn_pool_grid.cc

```diff
@@ -201,17 +225,22 @@ ConnectionPool::Cancellable* ConnectivityGrid::newStream(Http::ResponseDecoder&
   if (pools_.empty()) {
     createNextPool();
   }
-
+  PoolIterator pool = pools_.begin();
+  if (is_http3_broken_) {
+    ENVOY_LOG(trace, "HTTP/3 is broken to host '{}', skipping.", describePool(**pool),
+              host_->hostname());
+    createNextPool();
+    pool = ++pool;
+  }
   // TODO(#15649) track pools with successful connections: don't always start at
```

Comment:
are you doing this TODO?

Reply:
I think you are - by skipping pool ahead above :-P  Functioanlly if QUIC is broken we start at H2

Reply:
LOL! I totally misread what the TODO was for. You're right. Removed.